wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.wear.compose.material.SwipeToDismissBox
+import androidx.wear.compose.material.SwipeToDismissKeys
+import com.arkivanov.decompose.Child
+import com.arkivanov.decompose.extensions.compose.jetpack.subscribeAsState
+import com.arkivanov.decompose.router.stack.ChildStack
+import com.arkivanov.decompose.value.Value
+
+/**
+ * Displays the [ChildStack] in [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ *
+ * @param stack a [ChildStack] to be displayed.
+ * @param onDismissed called when the swipe to dismiss gesture has completed, allows popping the stack.
+ * See [StackNavigator#pop][com.arkivanov.decompose.router.stack.pop].
+ * @param modifier a [Modifier] to be applied to the underlying
+ * [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ * @param content a Composable slot displaying a [Child][Child.Created].
+ */
+@Composable
+fun <C : Any, T : Any> SwipeToDismissBox(
+    stack: Value<ChildStack<C, T>>,
+    onDismissed: () -> Unit,
+    modifier: Modifier = Modifier,
+    content: @Composable (child: Child.Created<C, T>) -> Unit,
+) {
+    val state = stack.subscribeAsState()
+
+    SwipeToDismissBox(
+        stack = state.value,
+        onDismissed = onDismissed,
+        modifier = modifier,
+        content = content,
+    )
+}
+
+/**
+ * Displays the [ChildStack] in [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ *
+ * @param stack a [ChildStack] to be displayed.
+ * @param onDismissed called when the swipe to dismiss gesture has completed, allows popping the stack.
+ * See [StackNavigator#pop][com.arkivanov.decompose.router.stack.pop].
+ * @param modifier a [Modifier] to be applied to the underlying
+ * [SwipeToDismissBox][androidx.wear.compose.material.SwipeToDismissBox].
+ * @param content a Composable slot displaying a [Child][Child.Created].
+ */
+@Composable
+fun <C : Any, T : Any> SwipeToDismissBox(
+    stack: ChildStack<C, T>,
+    onDismissed: () -> Unit,
+    modifier: Modifier = Modifier,
+    content: @Composable (child: Child.Created<C, T>) -> Unit,
+) {
+    val active: Child.Created<C, T> = stack.active
+    val background: Child.Created<C, T>? = stack.backStack.lastOrNull()
+    val holder = rememberSaveableStateHolder()
+
+    RetainStates(holder, stack.getConfigurations())
+
+    SwipeToDismissBox(
+        onDismissed = onDismissed,
+        modifier = modifier,
+        backgroundScrimColor = workaroundBackgroundScrimColor(stackSize = stack.items.size),
+        backgroundKey = background?.configuration ?: SwipeToDismissKeys.Background,
+        contentKey = active.configuration,
+        hasBackground = background != null,
+    ) { isBackground ->
+        val child = background?.takeIf { isBackground } ?: active
+        holder.SaveableStateProvider(child.configuration.key()) {
+            content(child)
+        }
+    }
+}
+
+// Workaround for https://issuetracker.google.com/issues/280392104
```

Comment:
This should be fixed in 1.3.0-alpha01 if we want to jump there

https://issuetracker.google.com/issues/280392104

Reply:
Yeah, @joreilly let me know if you are ok to use the alpha version.

Reply:
I think we are ok with it.

Reply:
@joreilly could you please review? 😀

Reply:
Let's land and follow up, I can bump and remove.